Sources/Swift/Helper/DataDeserialization.swift

```diff
@@ -29,4 +31,154 @@
+
+                    guard let type = headerDictionary?["type"] as? String else {
+                        SentrySDKLog.error("Envelope item type is required.")
+                        break
+                    }
+
+                    let bodyLength = (headerDictionary?["length"] as? NSNumber)?.uintValue ?? 0
+
+                    if i == endOfEnvelope && bodyLength != 0 {
+                        SentrySDKLog.error("Envelope item has no data but header indicates its length is \(bodyLength).")
+                        break
+                    }
+
+                    let filename = headerDictionary?["filename"] as? String
+                    let contentType = headerDictionary?["content_type"] as? String
+                    let attachmentType = headerDictionary?["attachment_type"] as? String
+                    let itemCount = headerDictionary?["item_count"] as? NSNumber
+
+                    let itemHeader: SentryEnvelopeItemHeader
+                    if let filename = filename {
+                        itemHeader = SentryEnvelopeAttachmentHeader(
+                            type: type,
+                            length: bodyLength,
+                            filename: filename,
+                            contentType: contentType,
+                            attachmentType: typeForSentryAttachmentName(attachmentType)
+                        )
+                    } else if let itemCount = itemCount {
+                        itemHeader = SentryEnvelopeItemHeader(
+                            type: type,
+                            length: bodyLength,
+                            contentType: contentType,
+                            itemCount: itemCount
+                        )
+                    } else {
+                        itemHeader = SentryEnvelopeItemHeader(type: type, length: bodyLength)
+                    }
+
+                    if i == endOfEnvelope {
+                        i += 1 // 0-byte attachment
+                    }
+
+                    if bodyLength > 0 && data.count < (i + 1 + Int(bodyLength)) {
+                        SentrySDKLog.error(
+                            "Envelope is corrupted or has invalid data. Trying to read \(bodyLength) bytes by skipping \(i + 1) from a buffer of \(data.count) bytes."
+                        )
+                        return nil
+                    }
+
+                    let itemBody = (data as NSData).subdata(with: NSRange(location: i + 1, length: Int(bodyLength)))
+                    let envelopeItem = SentryEnvelopeItem(header: itemHeader, data: itemBody)
+                    items.append(envelopeItem)
+
+                    i = i + 1 + Int(bodyLength)
+                    itemHeaderStart = i
+                } catch {
+                    SentrySDKLog.error("Failed to parse envelope item header \(error)")
+                    return nil
+                }
```

Comment:
`l`: Again I think we should be able to move this catch statement up a bit so it's closer to the try.